### ISSUE TITLE
'double CAT applied' warning on pristine image

### ISSUE THREAD

**Issue by kofa73** [darktable-org/darktable#7271]
**Describe the bug**
"double CAT applied" warning is shown for freshly opened image

**To Reproduce**
1. Open raw image (I tried with Nikon D7000 NEF and Panasonic LX7 RW2) in darkroom (clear history before, if needed)
2. Check the module list (screenshot is heavily cropped). Note that 'colour calibration' displays a warning.
![image](https://user-images.githubusercontent.com/3177739/101903048-f2957d00-3bb3-11eb-8277-f1628729779c.png)
3. Open 'colour calibration' to check warning:
![image](https://user-images.githubusercontent.com/3177739/101903143-1c4ea400-3bb4-11eb-8ac3-b990b5a737fa.png)
4. Check 'white balance' to ensure it's set to 'camera reference'
![image](https://user-images.githubusercontent.com/3177739/101903254-4a33e880-3bb4-11eb-8682-b6d5b175b11b.png)
5. Compress the history from item 4 (white balance, before 'colour calibration'
![image](https://user-images.githubusercontent.com/3177739/101903456-9aab4600-3bb4-11eb-8e97-49aee7f5e1dc.png)
6. Observe that colour calibration is re-added, with no 'double CAT' warning:
![image](https://user-images.githubusercontent.com/3177739/101903621-da722d80-3bb4-11eb-8875-e52863209767.png)

**Expected behavior**
No 'double CAT' warning when image is opened

**Platform (please complete the following information):**
 - Darktable Version: 3.3.0+2075~gc06b55abd
 - OS: Ubuntu 20.10

**Comment by aurelienpierre** [comment]
I suppose this does not happen with the first image you open, right ? You need to open a couple of them before that pops up ?

**Comment by Mark-64** [comment]
Confirmed here, latest git master on Windows 10, it happens even with the first image opened, starting with an empty database

**Comment by kofa73** [comment]
I get them immediately (image already in darktable's DB, darktable opened freshly, image history discarded, imaged opened). There seems to be some non-deterministic behaviour, e.g. one of my Nikon D7000 images came up with a greenish tint at first. However, I have been unable to reproduce that.

**Comment by kofa73** [comment]
I still have the issue with 3.3.0+2096~g528512d78
I built it today, downloaded the NEF from https://discuss.pixls.us/t/moire-nightmare-challenge-from-hell/21787 and the double CAT warning is still there.
![image](https://user-images.githubusercontent.com/3177739/102080080-223fc180-3e0e-11eb-8132-2b9e300e2aa4.png)

**Comment by johnny-bit** [comment]
I can confirm that this is still an issue.

pixel workflow defaults set to scene refered
chromatic adaptation defaults set to modern

this is darktable 3.3.0+2103~g0c237c9e2

tried the mendtioned image (with no XMP) and just imported it - and got warning about double CAT...

This is generated xmp with NOTHING done to it: [20201213_0022.NEF.xmp](https://github.com/darktable-org/darktable/files/5689026/20201213_0022.NEF.xmp.txt)

running darktable with `-d params` produces following output: [20201213_0022.params.txt](https://github.com/darktable-org/darktable/files/5689053/20201213_0022.params.txt)

Sorry @aurelienpierre, still not fixed yet ;(

**Comment by Mark-64** [comment]
I see another issue maybe connected.
Starting with commit  ab91bf320e4bba2b54ea525a988176d4735b7beb, when adaptation is set to none (bypass) the refresh of the central image preview in darkroom has become super slow, 5-6 second, while before it was almost instantaneous.
With other adaptation modes the refresh is normal. Bypassing CAT should be quicker, no ?

I am not sure whether to open a separate issue for this

**Comment by kofa73** [comment]
@aurelienpierre @TurboGit Please don't be angry :) It's still there with 3.3.0+2110~g58dfe2ff1 58dfe2ff1
![image](https://user-images.githubusercontent.com/3177739/102139643-66f34900-3e5e-11eb-8142-cb35b4782699.png)

I discarded the history of the image, then removed the image from darktable, imported it again. I also reproduced it with other images, from my other camera.

I thought the following would fix it:
```
commit bc21612fac0249ee94a07aca0c4d378569cd1071
Merge: f8801f5ce 061074e36
Author: Pascal Obry <pascal@obry.net>
Date:   Mon Dec 14 18:19:25 2020 +0100

    Merge pull request #7320 from aurelienpierre/fix-cat-warning
    
    fix false positive warning for CAT applied twice
```

**Comment by TurboGit** [comment]
Fact is that I cannot reproduce, I have never been able to reproduce :( Do you reproduce this 100% of the time ? Have you tried with latest master with a clean `darktablerc` ? Do you have any auto-preset applied ? Maybe demosaic that I see at the top of the history ? Any other ?

**Comment by TurboGit** [comment]
Note that we have only one more day for fixing this issue for 3.4.

**Comment by kofa73** [comment]
I can give it a try tomorrow

**Comment by johnny-bit** [comment]
I can reproduce 100% of the time

simple reproduction steps:

1. make sure the image is not in darktable DB, aslo make sure it doesn't have xmp file to it.
2. in lighttable click import -> image
3. darkroom opens with CAT warning :/

**Comment by kofa73** [comment]
I can also reproduce 100% of the time.
1. I removed ~/.cache/darktable; moved ~/.config/darktable out of the way.
2. I removed the XMP sidecar of the image
3. started darktable with `darktable -d opencl -d perf`
4. settings:
  - general/prefer performance over quality (was turned ON): turned OFF (was that turned on by default because this is an old machine?)
  - modify selected theme with CSS tweaks below (was turned ON): turned OFF
  - processing:
    - set workflow to 'scene-referred'
    - set chroma adaptation to 'modern'
    - set auto apply sharpen to 'false'
5. restarted darktable
6. imported the image
7. the 'double CAT applied' warning is present

I guess if this is just a bogus warning, and it does not affect processing quality, I can live with it. Switching to the lighttable and back to darkroom clears the warning.

There are a bunch of other settings marked as 'modified' (having the dot), even though I did not change them:
- darkroom: demosaicing for zoomed out darkroom mode: always bilinear (fast)
- cpu / gpu / memory:
  - number of background threads = 1
  - host memory limit for tiling: 500
  - minimum amount of memory for a single buffer: 8

**Comment by TurboGit** [comment]
> * make sure the image is not in darktable DB, aslo make sure it doesn't have xmp file to it.
> 
> * in lighttable click import -> image
> 
> * darkroom opens with CAT warning :/

I did just that, a new image, imported, no xmp of course in this case and no double CAT :( As I said probably something else somewhere...

**Comment by TurboGit** [comment]
Ok, tried with a fully clean dt, new db and clean darktablerc, swicth on scene-refered for the workflow and modern color calibration and did reproduce, so probably something else on my original config makes this non-reproducible :(

**Comment by Mark-64** [comment]
With the last commits I have the opposite problem, false negatives.
Tried in Windows and Ubuntu 20.04, starting with a clean database.
Creating a new instance of color calibration wit CAT gives no "double CAT" warning

**Comment by TurboGit** [comment]
Should be fixed (crossing fingers) in #7328
